tests/cpp/test_combined_inner_outer_reduction.cpp

```diff
@@ -34,255 +34,129 @@ namespace nvfuser {
-    fusion.addInput(grad_out);
-    fusion.addInput(input);
-    fusion.addInput(mean);
-    fusion.addInput(rstd);
-    fusion.addInput(weight);
-    fusion.addInput(bias);
-
-    if (dtype == DataType::Half) {
-      grad_out = castOp(DataType::Float, grad_out);
-      input = castOp(DataType::Float, input);
-      weight = castOp(DataType::Float, weight);
-      bias = castOp(DataType::Float, bias);
-    }
-
-    auto layer_norm_results = layer_norm_backward(
-        grad_out,
-        input,
-        norm_shape,
-        mean,
-        rstd,
-        weight,
-        bias,
-        {true, true, true});
-
-    if (dtype == DataType::Half) {
-      layer_norm_results.grad_input =
-          castOp(dtype, layer_norm_results.grad_input);
-      layer_norm_results.grad_bias =
-          castOp(dtype, layer_norm_results.grad_bias);
-      layer_norm_results.grad_weight =
-          castOp(dtype, layer_norm_results.grad_weight);
-    }
-
-    fusion.addOutput(layer_norm_results.grad_input);
-    fusion.addOutput(layer_norm_results.grad_weight);
-    fusion.addOutput(layer_norm_results.grad_bias);
-
-    auto maybe_fp16_options = at::TensorOptions()
-                                  .dtype(data_type_to_aten(dtype))
-                                  .device(at::kCUDA, 0);
-
-    at::Tensor aten_grad_out = at::randn(input_shape, maybe_fp16_options);
-    at::Tensor aten_input = at::randn(input_shape, maybe_fp16_options);
-    at::Tensor aten_weight = at::randn(norm_shape, maybe_fp16_options);
-    at::Tensor aten_bias = at::randn(norm_shape, maybe_fp16_options);
-
-    auto at_weight = c10::optional<at::Tensor>(aten_weight);
-    auto at_bias = c10::optional<at::Tensor>(aten_bias);
-
-    const float kEps = 1e-5;
-    auto aten_results =
-        at::native_layer_norm(aten_input, norm_shape, at_weight, at_bias, kEps);
-    auto aten_output = std::get<0>(aten_results);
-    auto aten_mean = std::get<1>(aten_results);
-    auto aten_rstd = std::get<2>(aten_results);
-
-    FusionExecutorCache fec(std::move(fusion_ptr));
-    std::vector<c10::IValue> aten_inputs = {
-        aten_grad_out,
-        aten_input,
-        aten_mean,
-        aten_rstd,
-        aten_weight,
-        aten_bias};
-    auto cg_outputs = fec.runFusionWithInputs(aten_inputs);
-
-    auto aten_gradients = at::native_layer_norm_backward(
-        aten_grad_out,
-        aten_input,
-        norm_shape,
-        aten_mean,
-        aten_rstd,
-        c10::optional<at::Tensor>(aten_weight),
-        c10::optional<at::Tensor>(aten_bias),
-        {true, true, true});
+// tuple of data type, batch size (outer dim), hidden size (inner dim)
+using CombinedSchedulerParams = std::tuple<DataType, int64_t, int64_t>;
+using CombinedSchedulerTest = NVFuserFixtureParamTest<CombinedSchedulerParams>;
+TEST_P(CombinedSchedulerTest, LayerNormBackward) {
+  std::unique_ptr<Fusion> fusion_ptr = std::make_unique<Fusion>();
+  Fusion& fusion = *fusion_ptr.get();
+  FusionGuard fg(&fusion);
```

Comment:
```suggestion
  auto fusion = std::make_unique<Fusion>();
  FusionGuard fg(fusion.get());
```
Then replace `fusion.` in this test with `fusion->`. 

Having two variables `fusion_ptr` and `fusion` in this case provide little value. So I prefer using one variable for simplicity -- readers don't even need to think about the difference between the two variables.